kolena/_experimental/object_detection/dataset.py

```diff
@@ -0,0 +1,246 @@
+        LabeledBoundingBox(
+            label=gt.label,
+            top_left=gt.top_left,
+            bottom_right=gt.bottom_right,
+            predicted_label=inf.label,  # type: ignore[call-arg]
+            predicted_score=inf.score,  # type: ignore[call-arg]
+        )
+        if inf
+        else gt
+        for gt, inf in bbox_matches.unmatched_gt
+    ]
+    confused = [
+        ScoredLabeledBoundingBox(
+            label=inf.label,
+            score=inf.score,
+            top_left=inf.top_left,
+            bottom_right=inf.bottom_right,
+            actual_label=gt.label,  # type: ignore[call-arg]
+        )
+        for gt, inf in bbox_matches.unmatched_gt
+        if inf is not None and inf.score >= thresholds[inf.label]
+    ]
+    scores = [inf.score for inf in tp + fp]
+    inference_labels = {inf.label for _, inf in bbox_matches.matched} | {
+        inf.label for inf in bbox_matches.unmatched_inf
+    }
+    fields = [
+        ScoredLabel(label=label, score=thresholds[label])
+        for label in sorted(thresholds.keys())
+        if label in inference_labels
+    ]
+    return dict(
+        TP=tp,
+        FP=fp,
+        FN=fn,
+        matched_inference=[inf for _, inf in bbox_matches.matched],
+        unmatched_ground_truth=unmatched_ground_truth,
+        unmatched_inference=bbox_matches.unmatched_inf,
+        Confused=confused,
+        count_TP=len(tp),
+        count_FP=len(fp),
+        count_FN=len(fn),
+        count_Confused=len(confused),
+        has_TP=len(tp) > 0,
+        has_FP=len(fp) > 0,
+        has_FN=len(fn) > 0,
+        has_Confused=len(confused) > 0,
+        max_confidence_above_t=max(scores) if len(scores) > 0 else None,
+        min_confidence_above_t=min(scores) if len(scores) > 0 else None,
+        thresholds=fields,
+    )
+
+
+def _compute_metrics(
+    pred_df: pd.DataFrame,
+    *,
+    ground_truth: str,
+    inference: str,
+    iou_threshold: float = 0.5,
+    threshold_strategy: Union[Literal["F1-Optimal"], float, Dict[str, float]] = 0.5,
+    min_confidence_score: float = 0.5,
+) -> pd.DataFrame:
+    """
+    Compute metrics for object detection.
+
+    :param df: Dataframe for model results.
+    :param ground_truth: Column name for ground_truth bounding boxes
+    :param inference: Column name for inference bounding boxes
+    :param iou_threshold: The [IoU ↗](../../metrics/iou.md) threshold, defaulting to `0.5`.
+    :param threshold_strategy: The confidence threshold strategy. It can either be a fixed confidence threshold such
+        as `0.5` or `0.75`, or the F1-optimal threshold.
+    :param min_confidence_score: The minimum confidence score to consider for the evaluation. This is usually set to
+        reduce noise by excluding inferences with low confidence score.
+    """
+    labels = {x.label for x in itertools.chain.from_iterable(pred_df[ground_truth])}
+    labels.union({x.label for x in itertools.chain.from_iterable(pred_df[inference])})
```

Comment:
i think union op is not in-place and we don't save the results to any variable

Reply:
might worth to update the test to cover this

Reply:
thanks for catching, updated and test added